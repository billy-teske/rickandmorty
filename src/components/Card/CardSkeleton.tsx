import styles from './CardSkeleton.module.css';
import listStyle from '../List/List.module.css';

const CardSkeleton = () => (
    <div className={listStyle.list}>
        {Array.from({ length: 20 }, (_, id) => (
            <div key={id} className={styles.skeleton} style={{
                animationDelay: `${id / 10}s`
            }}></div>
        ))}
    </div>
);

export default CardSkeleton;
