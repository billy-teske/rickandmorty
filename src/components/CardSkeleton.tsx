import { useEffect, useState } from 'react';
import styles from './CardSkeleton.module.css';
import listStyle from './List.module.css';

const CardSkeleton = () => {
    const [arrayCount, setArrayCount] = useState([0]);

    useEffect(() => {
        const intervalId = setInterval(() => {
            setArrayCount((prevCount) => [prevCount.length, ...prevCount]);
        }, 700);
        return () => clearInterval(intervalId);
    }, []);

    return (
        <div className={listStyle.list}>
            {arrayCount.map((id) => (
                <div key={id} className={styles.skeleton}></div>
            ))}
        </div>
    );
};

export default CardSkeleton;
