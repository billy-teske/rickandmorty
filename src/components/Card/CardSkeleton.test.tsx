import { render } from '@testing-library/react';
import CardSkeleton from './CardSkeleton';

describe('CardSkeleton Component', () => {
    it('should render a cardSkeleton', () => {
        const result = render(
            <CardSkeleton />
        );

        expect(result).toMatchSnapshot();
    });
});