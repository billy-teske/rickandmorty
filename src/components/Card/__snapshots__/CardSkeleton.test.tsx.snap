// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSkeleton Component should render a cardSkeleton 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="list"
      >
        <div
          class="skeleton"
          style="animation-delay: 0s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 0.1s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 0.2s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 0.3s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 0.4s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 0.5s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 0.6s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 0.7s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 0.8s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 0.9s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 1s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 1.1s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 1.2s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 1.3s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 1.4s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 1.5s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 1.6s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 1.7s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 1.8s;"
        />
        <div
          class="skeleton"
          style="animation-delay: 1.9s;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="list"
    >
      <div
        class="skeleton"
        style="animation-delay: 0s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 0.1s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 0.2s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 0.3s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 0.4s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 0.5s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 0.6s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 0.7s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 0.8s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 0.9s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 1s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 1.1s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 1.2s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 1.3s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 1.4s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 1.5s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 1.6s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 1.7s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 1.8s;"
      />
      <div
        class="skeleton"
        style="animation-delay: 1.9s;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
